proxy/src/control/observe.rs

```diff
@@ -115,21 +122,21 @@ impl Stream for TapEvents {
                                 continue;
                             }
                             self.remaining -= 1;
-                            let _ = self.current.insert(req.clone(), ());
+                            let _ = self.current.insert(RequestById { req: req.clone() });
```

Comment:
TIOLI: making `RequestById` a tuple struct might make this slightly simpler (`RequestById(req.clone())`)

Reply:
Done!